C generated code contains an error

### Describe the bug

The C generated code contains an error.
V code :
![image](https://user-images.githubusercontent.com/68387948/216820220-c17f8805-fb67-48bc-88af-b687f5dba737.png)
![image](https://user-images.githubusercontent.com/68387948/216820245-e5449746-c71a-44d5-8181-95d8b2468260.png)

Result :
![image](https://user-images.githubusercontent.com/68387948/216820301-50cfcb5e-0cc3-42d2-9b87-6351867a7869.png)

Generated C code that contains an error :
![image](https://user-images.githubusercontent.com/68387948/216820362-35d6f6f0-5c64-4238-92d4-c785c7c076d9.png)


### Expected Behavior

The code to compile.

### Current Behavior
```
/home/riccardo/v/v /home/riccardo/V_orm/FirstProject -color -o /home/riccardo/V_orm/FirstProject/bin/main
builder error: 
==================
C error. This should never happen.
This is a compiler bug, please report it using `v bug file.v`.
https://github.com/vlang/v/issues/new/choose
You can also use #help on Discord: https://discord.gg/vlang
==================
/tmp/v_1000/main.7252778447995485324.tmp.c:25966: error: cannot convert 'struct net__TcpConn *' to 'struct net__TcpConn'
...
==================
(Use `v -cg` to print the entire error message)
```
### Reproduction Steps
```v
module main

import net

fn main() {
	ignore := net.TcpConn{}
	mut array := []&net.TcpConn{}

	for mut socket in array {
		//Error is here
		if ignore!=socket {

		}
	}
}
```



### Possible Solution

**I just had to replace "mut sockets []&net.TcpConn" by "mut sockets []net.TcpConn" everywhere.**

### Additional Information/Context

_No response_

### V version

V 0.3.3 858ce4e

### Environment details (OS name and version, etc.)

OS: linux, Ubuntu 22.10
Processor: 2 cpus, 64bit, little endian, Intel(R) Celeron(R) N4020 CPU @ 1.10GHz
CC version: cc (Ubuntu 12.2.0-3ubuntu1) 12.2.0

getwd: /home/riccardo
vmodules: /home/riccardo/.vmodules
vroot: /home/riccardo/v
vexe: /home/riccardo/v/v
vexe mtime: 2023-02-05 13:07:14
is vroot writable: true
is vmodules writable: true
V full version: V 0.3.3 bb512f7.858ce4e

Git version: git version 2.37.2
Git vroot status: weekly.2023.05-9-g858ce4e3
.git/config present: true
thirdparty/tcc status: thirdparty-linux-amd64 12f392c3

--- a/vlib/v/gen/c/infix.v
+++ b/vlib/v/gen/c/infix.v
@@ -144,12 +144,12 @@ fn (mut g Gen) infix_expr_eq_op(node ast.InfixExpr) {
 				}
 				g.write('${ptr_typ}_alias_eq(')
 				if left.typ.is_ptr() {
-					g.write('*')
+					g.write('*'.repeat(left.typ.nr_muls()))
 				}
 				g.expr(node.left)
 				g.write(', ')
 				if right.typ.is_ptr() {
-					g.write('*')
+					g.write('*'.repeat(right.typ.nr_muls()))
 				}
 				g.expr(node.right)
 				g.write(')')
@@ -161,7 +161,7 @@ fn (mut g Gen) infix_expr_eq_op(node ast.InfixExpr) {
 				}
 				g.write('${ptr_typ}_arr_eq(')
 				if left.typ.is_ptr() && !left.typ.has_flag(.shared_f) {
-					g.write('*')
+					g.write('*'.repeat(left.typ.nr_muls()))
 				}
 				g.expr(node.left)
 				if left.typ.has_flag(.shared_f) {
@@ -169,7 +169,7 @@ fn (mut g Gen) infix_expr_eq_op(node ast.InfixExpr) {
 				}
 				g.write(', ')
 				if right.typ.is_ptr() && !right.typ.has_flag(.shared_f) {
-					g.write('*')
+					g.write('*'.repeat(right.typ.nr_muls()))
 				}
 				g.expr(node.right)
 				if right.typ.has_flag(.shared_f) {
@@ -210,36 +210,32 @@ fn (mut g Gen) infix_expr_eq_op(node ast.InfixExpr) {
 				}
 				g.write('${ptr_typ}_map_eq(')
 				if left.typ.is_ptr() {
-					g.write('*')
+					g.write('*'.repeat(left.typ.nr_muls()))
 				}
 				g.expr(node.left)
 				g.write(', ')
 				if right.typ.is_ptr() {
-					g.write('*')
+					g.write('*'.repeat(right.typ.nr_muls()))
 				}
 				g.expr(node.right)
 				g.write(')')
 			}
 			.struct_ {
-				// if g.pref.translated {
-				// g.gen_plain_infix_expr(node)
-				//} else {
 				ptr_typ := g.equality_fn(left.unaliased)
 				if node.op == .ne {
 					g.write('!')
 				}
 				g.write('${ptr_typ}_struct_eq(')
 				if left.typ.is_ptr() {
-					g.write('*')
+					g.write('*'.repeat(left.typ.nr_muls()))
 				}
 				g.expr(node.left)
 				g.write(', ')
 				if right.typ.is_ptr() {
-					g.write('*')
+					g.write('*'.repeat(right.typ.nr_muls()))
 				}
 				g.expr(node.right)
 				g.write(')')
-				//}
 			}
 			.sum_type {
 				ptr_typ := g.equality_fn(left.unaliased)
@@ -248,12 +244,12 @@ fn (mut g Gen) infix_expr_eq_op(node ast.InfixExpr) {
 				}
 				g.write('${ptr_typ}_sumtype_eq(')
 				if left.typ.is_ptr() {
-					g.write('*')
+					g.write('*'.repeat(left.typ.nr_muls()))
 				}
 				g.expr(node.left)
 				g.write(', ')
 				if right.typ.is_ptr() {
-					g.write('*')
+					g.write('*'.repeat(right.typ.nr_muls()))
 				}
 				g.expr(node.right)
 				g.write(')')
@@ -265,12 +261,12 @@ fn (mut g Gen) infix_expr_eq_op(node ast.InfixExpr) {
 				}
 				g.write('${ptr_typ}_interface_eq(')
 				if left.typ.is_ptr() {
-					g.write('*')
+					g.write('*'.repeat(left.typ.nr_muls()))
 				}
 				g.expr(node.left)
 				g.write(', ')
 				if right.typ.is_ptr() {
-					g.write('*')
+					g.write('*'.repeat(right.typ.nr_muls()))
 				}
 				g.expr(node.right)
 				g.write(')')

--- a/vlib/v/tests/for_in_mut_array_with_infix_expr_test.v
+++ b/vlib/v/tests/for_in_mut_array_with_infix_expr_test.v
@@ -1,0 +1,13 @@
+import net
+
+fn test_for_in_mut_array_with_infix_expr() {
+	ignore := net.TcpConn{}
+	mut array := []&net.TcpConn{}
+
+	for mut socket in array {
+		if ignore != socket {
+			println('not eq')
+		}
+	}
+	assert true
+}

